app/features/dynamo/tools.py

```diff
@@ -1,21 +1,94 @@
 from langchain_community.document_loaders import YoutubeLoader
+from langchain_community.document_loaders import PyPDFLoader
+from langchain_core.documents import Document
+from langchain_community.document_loaders.csv_loader import CSVLoader
+from langchain_community.document_loaders import TextLoader
+from langchain_community.document_loaders import UnstructuredURLLoader
+from langchain_community.document_loaders import UnstructuredPowerPointLoader
+from langchain_community.document_loaders import Docx2txtLoader
+from langchain_community.document_loaders import UnstructuredExcelLoader
+from langchain_community.document_loaders import UnstructuredXMLLoader
+from langchain_google_genai import ChatGoogleGenerativeAI
+from app.utils.allowed_file_extensions_dynamo import FileType
+from app.api.error_utilities import FileHandlerError, ImageHandlerError
 from langchain_text_splitters import RecursiveCharacterTextSplitter
 from langchain.prompts import PromptTemplate
 from langchain_google_genai import GoogleGenerativeAI
 from langchain_core.output_parsers import JsonOutputParser
 from langchain.chains.summarize import load_summarize_chain
 from langchain_core.pydantic_v1 import BaseModel, Field
 from app.api.error_utilities import VideoTranscriptError
+from langchain_core.messages import HumanMessage
 from fastapi import HTTPException
 from app.services.logger import setup_logger
 import os
+import tempfile
+import uuid
+import requests
+import gdown
 
+STRUCTURED_TABULAR_FILE_EXTENSIONS = {"csv", "xls", "xlsx", "gsheet", "xml"}
 
 logger = setup_logger(__name__)
 
 # AI Model
 model = GoogleGenerativeAI(model="gemini-1.0-pro")
 
+splitter = RecursiveCharacterTextSplitter(
+    chunk_size = 1000,
+    chunk_overlap = 0
+)
+
+def build_chain(prompt: str):
+    prompt_template = read_text_file(prompt)
+    summarize_prompt = PromptTemplate.from_template(prompt_template)
+
+    summarize_model = GoogleGenerativeAI(model="gemini-1.5-flash")
+        
+    chain = summarize_prompt | summarize_model 
+    return chain
+
+def get_summary(file_url: str, file_type: str, verbose=True):
+    file_type = file_type.lower()
+    try:
+        file_loader = file_loader_map[FileType(file_type)]
+        full_content = file_loader(file_url, verbose)
+        if file_type in STRUCTURED_TABULAR_FILE_EXTENSIONS:
+            prompt = "prompt/summarize-structured-tabular-data-prompt.txt"
```

Comment:
Always prefix file paths with an `r` when importing files to prevent issues with `/` across different platforms:

```python
prompt = r"prompt/summarize-structured-tabular-data-prompt.txt"
```

Reply:
Done.